content/spin/v3/writing-apps.md

```diff
@@ -0,0 +1,530 @@
+
+```toml
+# Use the `security:http` package in the default registry
+"security:http/malice" = "2.0.0"
+```
+
+#### Dependencies from a Local Component
+
+To use a dependency from a component on your file system, specify the following fields:
+
+| Field      | Required?   | Description                                                                                  | Example |
+|------------|-------------|----------------------------------------------------------------------------------------------|---------|
+| `path`     | Required    | The path to the Wasm file containing the component.                                          | `"../validation/request-checker.wasm"` |
+| `export`   | Optional    | The name of the export in the package. If omitted, this defaults to the name of the import.  | `"more-secure:checking-it-out/web"` |
+
+#### Dependencies from a URL
+
+To use a dependency from an HTTP URL, such as a GitHub release, specify the following fields:
+
+| Field      | Required?   | Description                                                                                  | Example |
+|------------|-------------|----------------------------------------------------------------------------------------------|---------|
+| `url`      | Required    | The URL of the Wasm file containing the component.                                           | `"https://example.com/downloads/request-checker.wasm"` |
+| `digest`   | Required    | The SHA256 digest of the Wasm file. This is required for integrity checking.                 | `"sha256:650376c33a0756b1a52cad7ca670f1126391b79050df0321407da9c741d32375"` |
+| `export`   | Optional    | The name of the export in the package. If omitted, this defaults to the name of the import.  | `"more-secure:checking-it-out/web"` |
+
+### Mapping All Imports from a Package
+
+If you are importing several interfaces from the same WIT package, and want them all satisfied by the same Wasm package, you can omit the interface from the dependency name. For example, suppose you import the `malice`, `tomfoolery`, and `shenanigans` WIT interfaces from the `security:http` package, and that your Bargain Security package exports all three of them.  You can write:
+
+```toml
+[component.my-app.dependencies]
+"security:http" = { package = "bargains:inspection", version = "2.0.0", registry = "packages.example.com" }
+```
+
+and Spin will map all of your `security:http` imports to the matching exports from the package.
+
+### Dependency Permissions
+
+By default, dependencies do not have access to Spin resources that require permission to be given in the manifest - network hosts, key-value stores, SQLite databases, variables, etc.
+
+If you depend on a component which requires such access, and you trust all the components that you depend on, you can grant them access to the same resources that the 'main' component has by setting the `dependencies_inherit_configuration` flag on the main component:
```

Comment:
'Main' was intended to refer to the component that the dependency is a dependency of.  Like the 'main' one in the, uh, component (composition), not the 'main' one in the application.  I'll try to rephrase but would welcome ideas!

Reply:
Okay I think this gets clearer if I talk about the 'component manifest' rather than a 'main' or 'parent' component e.g.

> If a component has a dependency which requires resource access, you can grant it by setting the `dependencies_inherit_configuration` flag in the Spin component manifest:
> ...
> This grants _all_ dependencies access to _all_ resources listed in the Spin component manifest. You should therefore set this only if you trust _all_ dependencies.

Reply:
I like that approach!